apps/web/src/components/layout/components/v2/SyncInfoModalTrigger.tsx

```diff
@@ -15,11 +52,65 @@ export function SyncInfoModalTrigger() {
 
   return (
     <>
-      <Button size="xs" Icon={IconOutlineCloudUpload} onClick={toggleSyncInfoModalShow}>
-        Sync
-      </Button>
+      <Tooltip
+        disabled={isSynced}
+        width="auto"
+        label={<Text>You have un-synced changes in your local application.</Text>}
+        withinPortal
+      >
+        <Container>
+          <Button size="xs" Icon={IconOutlineCloudUpload} onClick={toggleSyncInfoModalShow}>
+            Sync
+          </Button>
+          <When truthy={!isSynced}>
+            <Dot />
+          </When>
+        </Container>
+      </Tooltip>
+
       {/** TODO: use a modal manager */}
-      <SyncInfoModal isOpen={showSyncInfoModal} toggleOpen={toggleSyncInfoModalShow} />
+      <SyncInfoModal isOpen={showSyncInfoModal} toggleOpen={toggleSyncInfoModalShow} refetchOriginWorkflows={refetch} />
     </>
   );
 }
+
+function createHash(workflowDefine) {
+  return CryptoJS.SHA256(JSON.stringify(workflowDefine || '')).toString(CryptoJS.enc.Hex);
+}
+
+export function GradientDot(props) {
+  const { colorScheme } = useColorScheme();
+
+  return (
+    <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16" fill="none" {...props}>
+      <rect
+        x="1.5"
+        y="1.5"
+        width="13"
+        height="13"
+        rx="6.5"
+        fill="url(#paint0_linear_1722_2699)"
+        stroke={colorScheme === 'light' ? colors.white : colors.B15}
+        strokeWidth="3"
+      />
+      <defs>
+        <linearGradient id="paint0_linear_1722_2699" x1="8" y1="13" x2="8" y2="3" gradientUnits="userSpaceOnUse">
+          <stop stopColor="#FF512F" />
+          <stop offset="1" stopColor="#DD2476" />
+        </linearGradient>
+      </defs>
+    </svg>
+  );
+}
+
+const Container = styled.div`
+  position: relative;
+`;
+
+const Dot = styled(GradientDot)`
+  position: absolute;
+  top: -3px;
+  right: -3px;
+  width: 12px;
+  height: 12px;
+`;
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
☑ **chore:** Please use Panda instead. If you need some help, please let me know!‏

Reply:
I will need some help. I wanted to ask somewhere if we have a document that explains the recent changes we've had because I couldn't find it in notion, I was out of scope. I'm not sure exactly how we currently operate regarding the styles and design altogether.

Reply:
done :)